src/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -686,6 +692,12 @@ export const ShippingRoute: FC<ShippingProps> = props => {
     return false
   }
 
+  useEffect(() => {
+    if (readyToSaveVerifiedAddress) {
+      finalizeFulfillment()
+    }
+  }, [readyToSaveVerifiedAddress])
+
```

Comment:
We wouldn't want to merge this as-is because this function does other things and the dependency array is intentionally missing `finalizeFulfillment`. That function could be added but then several more would need to be wrapped in `useCallback()`

Reply:
I don't think its imperative that we pass in finalize fulfillment and then callbackize everything. We know what we're doing -- check to see if that boolean state has changed -- and so its fine to add a exaustive deps ignore (imo)

Reply:
Assuming this change is verified by @starsirius (or someone else) we should really get it shipped asap since this UX issue certainly has the potential to kill a sale.

Reply:
Agree that passing an arg could work, but threading state through is likely more complex vs setting (this) state to trigger an effect, since this is scoped to exactly one location setting the state, and nothing else needs to change.